routers/api/v1/user/watch.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2016 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package user
+
+import (
+	"time"
+
+	api "code.gitea.io/sdk/gitea"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/setting"
+)
+
+// WatchInfo contains information about a watched repository
+type WatchInfo struct {
+	Subscribed    bool        `json:"subscribed"`
+	Ignored       bool        `json:"ignored"`
+	Reason        interface{} `json:"reason"`
+	CreatedAt     time.Time   `json:"created_at"`
+	URL           string      `json:"url"`
+	RepositoryURL string      `json:"repository_url"`
+}
+
+// getWatchedRepos returns the repos that the user with the specified userID is
+// watching
+func getWatchedRepos(userID int64, private bool) ([]*api.Repository, error) {
+	watchedRepos, err := models.GetWatchedRepos(userID, private)
+	if err != nil {
+		return nil, err
+	}
+	user, err := models.GetUserByID(userID)
+	if err != nil {
+		return nil, err
+	}
+
+	repos := make([]*api.Repository, len(watchedRepos))
+	for i, watched := range watchedRepos {
+		access, err := models.AccessLevel(user, watched)
+		if err != nil {
+			return nil, err
+		}
+		repos[i] = watched.APIFormat(&api.Permission{
```

Comment:
If you want this to work with your api/fixes branch, need to change this to just watched.APIFormat(access)